staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go

```diff
@@ -71,6 +71,9 @@ func (a *mutatingDispatcher) Dispatch(ctx context.Context, attr *generic.Version
 			}
 			klog.Warningf("Failed calling webhook, failing closed %v: %v", hook.Name, err)
 		}
+		if statusErr, ok := err.(*apierrors.StatusError); ok {
+			return statusErr
+		}
 		return apierrors.NewInternalError(err)
```

Comment:
just returning err should be fine here. b/c non-status error will be defaulted to internal error

/cc @liggitt

Reply:
true, validating webhook just returns raw errors. in the ErrCallingWebhook case, it does wrap in NewInternalError

Reply:
This is related: https://github.com/kubernetes/kubernetes/pull/69520

Reply:
validation webhook passes all non-ErrCallingWebhook errors through as is. should we not do the same here?

Reply:
Will a 404 of the webhook result in a 404 for the request sending client? Do we want that?

Reply:
from reading the code, errors calling the webhook come back as ErrCallingWebhook errors, so no:

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go#L114-L116

Reply:
So we should just `return err` here? I will take a look to understand this..

Reply:
no, I'm saying I think the code is correct as-is in this PR. We won't return a NotFound error to the user if we failed calling the webhook

Reply:
If I understand correctly:

1. If it is `webhook.ErrCallingWebhook`, then we should wrap as `NewInternalError(err)`.
1. Otherwise return the raw error.